proposals/xxxx-weburl.md

```diff
@@ -0,0 +1,795 @@
+guide: [Using WebURL with Foundation][docs-foundationinterop].
+  
+### Would you believe there's more?
+
+We've already covered a lot, and there are still great APIs that don't _quite_ make this discussion but are worth checking out anyway.
+
+- The JavaScript `URL` API.
+
+  The JavaScript `URL` class API is also defined by the URL standard, meaning WebURL is able to offer a truly compatible interface
+  via its [`.jsModel` view][docs-jsmodel]. The string-typed URL components are already mostly the same (we trim some delimiters, that's it),
+  but JavaScript has some quirks that aren't desirable in WebURL:
+
+  ```javascript
+  // JavaScript assignment allows trailing garbage:
+
+  var url      = new URL("https://example.com/");
+  url.protocol = "ftp://foobar.org/";
+  url.href;
+  // "ftp://example.com"
+  //  ^^^ - Set the scheme and just ignored the rest. Not even an error.
+  ```
+
+  WebURL's native API is designed to be a better choice for Swift developers, but if you're creating content that will primarily
+  be consumed by JavaScript, it _can be helpful_ to have the exact same API available in Swift (especially if it includes the delimiters and other quirks).
+  With WebURL, you can have that.
+
+- Swift-system integration
+
+  WebURL includes conversions between swift-system's `FilePath` and `file:` URLs for both POSIX and Windows-style paths.
+  This conversion is not standardized anywhere, there are several issues with Foundation's implementation,
+  and a lot of compatibility quirks to consider. For WebURL, I devised a new implementation and [extensive test-suite][weburl-filepath-tests],
+  based on an analysis of Chromium, IE8, Firefox, WebKit, Rust, and the Windows shell APIs.
+
+  It took a long time to get it right!
+
+### IDNA
+
+WebURL is currently lacking one feature before it has full compatibility with the WHATWG URL Standard: Internationalized Domain Names (IDNA).
+
+The reason for this is that IDNA requires Unicode normalization, and the required data has not been made available by the standard library yet.
+Being "pure Swift" and not having any dependencies beyond the standard library are important goals for WebURL, so a dependency on ICU
+to fill the void is not acceptable. Any inputs that would require IDNA to process, are treated as invalid.
+
+I consider it a very important feature to add, although it will likely arrive some time after v1.0 and overall API stability.
+Our current URL type also has poor support for IDNA (`URL(string: "https://👁️👄👁️.fm")` also returns `nil`), so it technically isn't a regression.
+
+### Summary
+
+WebURL is a new URL type, designed to improve URL processing in Swift on all fronts:
+
+- It has modern **standards support**, bringing web-compatible URL processing to Swift for the first time,
+  and matching browsers and more libraries in other languages.
+
+- Its API is truly **designed for Swift**, and leverages the language to great effect: the `Host` enum gives you greater
```

Comment:
this is the most important paragraph, and should be moved to the front. in fact, i think the summary should come first…